qiskit/scheduler/methods/lowering.py

```diff
@@ -0,0 +1,130 @@
+from qiskit.pulse.exceptions import PulseError
+from qiskit.pulse.schedule import Schedule
+from qiskit.pulse import instructions as pulse_inst
+
+from qiskit.scheduler.config import ScheduleConfig
+from qiskit.scheduler.utils import measure
+
+
+CircuitPulseDef = namedtuple('CircuitPulseDef', [
+    'schedule',  # The schedule which implements the quantum circuit command
+    'qubits'])   # The labels of the qubits involved in the command according to the circuit
+
+
+def lower_gates(circuit: QuantumCircuit, schedule_config: ScheduleConfig) -> List[CircuitPulseDef]:
+    """
+    Return a list of Schedules and the qubits they operate on, for each element encountered in th
+    input circuit.
+
+    Without concern for the final schedule, extract and return a list of Schedules and the qubits
+    they operate on, for each element encountered in the input circuit. Measures are grouped when
+    possible, so ``qc.measure(q0, c0)`` or ``qc.measure(q1, c1)`` will generate a synchronous
+    measurement pulse.
+
+    Args:
+        circuit: The quantum circuit to translate.
+        schedule_config: Backend specific parameters used for building the Schedule.
+
+    Returns:
+        A list of CircuitPulseDefs: the pulse definition for each circuit element.
+
+    Raises:
+        QiskitError: If circuit uses a command that isn't defined in config.inst_map.
+    """
+    circ_pulse_defs = []
+
+    inst_map = schedule_config.inst_map
+    qubit_mem_slots = {}  # Map measured qubit index to classical bit index
+
+    def get_measure_schedule() -> CircuitPulseDef:
+        """Create a schedule to measure the qubits queued for measuring."""
+        sched = Schedule()
+        sched += measure(qubits=list(qubit_mem_slots.keys()),
+                         inst_map=inst_map,
+                         meas_map=schedule_config.meas_map,
+                         qubit_mem_slots=qubit_mem_slots)
+        qubit_mem_slots.clear()
+        return CircuitPulseDef(schedule=sched,
+                               qubits=[chan.index for chan in sched.channels
+                                       if isinstance(chan, AcquireChannel)])
+
+    for inst, qubits, clbits in circuit.data:
+        inst_qubits = [qubit.index for qubit in qubits]  # We want only the indices of the qubits
+        if any(q in qubit_mem_slots for q in inst_qubits):
+            # If we are operating on a qubit that was scheduled to be measured, process that first
+            circ_pulse_defs.append(get_measure_schedule())
+
+        if isinstance(inst, Barrier):
+            circ_pulse_defs.append(CircuitPulseDef(schedule=inst, qubits=inst_qubits))
+
+        elif isinstance(inst, Delay):
+            if inst.unit == 'dt':
+                duration = inst.duration
+            else:
+                duration_s = _convert_to_seconds(inst.duration, inst.unit)
+                duration = int(duration_s // schedule_config.dt)
```

Comment:
I think maybe we should only allow `dt` unit delays in scheduling. So if `inst.unit != 'dt'`, raise an error and say that the circuit needs to be transpiled. This would sort of follow the pattern we've established for basis gates as well -- the user must only schedule circuits that are written in terns of the basis gates of the device they're targeting. The difference, I guess, is that a user could add new gates with an expanded `inst_map`.

@chriseclectic , @ajavadia ? What do you guys think? It would allow us to delete this out-of-place and duplicated logic of `_convert_to_seconds` method in this file, and also allow us to remove the newly added dependency of the `scheduler` on `dt`.

Reply:
I've put all logics for dt conversion into one pass (DelayInDt) at fe03e16